daml-lf/spec/daml-lf-1.rst

```diff
@@ -1653,7 +1686,101 @@ bound by pattern.
     —————————————————————————————————————————————————————————————————————— MatchFail
        v 'matches' p  ⇝  Fail
 
+       
+Value equality
+~~~~~~~~~~~~~~
+
+We define here the relation ``~ᵥ`` on values that is used as equality
+check. This is a partial equivalence relation over all values, but a
+(total) equivalence relation over serialized values. This relation
+will always be used to compare values of same types::
 
+                                  ┌────────┐
+  Generic Equivalence Relation    │ v ~ᵥ w │
+                                  └────────┘
+	   
+   LitNumeric₁ and LitNumeric₂ have the same scale and value
+  ——————————————————————————————————————————————————— GenEqLitNumeric
+   LitNumeric₁ ~ᵥ LitNumeric₂
+
+   t₁ and t₂ have the same value
+  ——————————————————————————————————————————————————— GenEqLitText
+   t₁ ~ᵥ t₂
+
+   LitDate₁ and LitDate₂ have the same value
+  ——————————————————————————————————————————————————— GenEqLitDate
+   LitDate₁ ~ᵥ LitDate₂
+
+   LitTimestamp₁ and LitTimestamp₂ have the same value
+  ——————————————————————————————————————————————————— GenEqLitTimestamp
+   LitTimestamp₁ ~ᵥ LitTimestamp₂
+
+   cid₁ and cid₂ are the same
+  ——————————————————————————————————————————————————— GenEqLitContractId
+   cid₁ ~ᵥ cid₂
+
+  ——————————————————————————————————————————————————— GenEqUnit
+   () ~ᵥ ()
+
+  ——————————————————————————————————————————————————— GenEqTrue
+   'True' ~ᵥ 'True'
+
+  ——————————————————————————————————————————————————— GenEqFalse
+   'False' ~ᵥ 'False'
+
+  ——————————————————————————————————————————————————— GenEqListNil
+   'Nil' @τ₁ ~ᵥ 'Nil' @τ₂
+
+   vₕ ~ᵥ wₕ  vₜ ~ᵥ wₜ
+  ——————————————————————————————————————————————————— GenEqListCons
+   'Cons' @τ vₕ vₜ  ~ᵥ 'Cons' @τ wₜ wₜ
+
+  ——————————————————————————————————————————————————— GenEqOptionNone
+   'None' @τ ~ᵥ 'None' @σ
+
+   v ~ᵥ w
+  ——————————————————————————————————————————————————— GenEqOptionSome
+   'Some' @τ v ~ᵥ 'Some' @τ w
```

Comment:
I think it makes sense to allow different types between the `'Some'` constructors, as we do for the `'None'` case and for other constructors, because types are erased. Even if we only ever compare values of the same type.

```suggestion
   'Some' @τ v ~ᵥ 'Some' @σ w
```

Reply:
fixed.